### PR TITLE
Feature/book list

### PR DESCRIPTION
**Pull request by blacksam07**
## Summary
- Add tokenInterceptor to add headers
- Add view to bookList
- Add book service 

## Trello Card 
[12-book-list-auth-header](https://trello.com/c/S13XU2Tj/12-book-list-auth-header)

## Screenshots

book_list-min

### CODE REVIEW

File: src/app/screens/book-list/book-list.component.html [hunk 1]
```diff
@@ -1 +1,9 @@
-<p>book-list works!</p>
+<div class="list-container">
+  <ng-template ngFor let-book [ngForOf]="books" let-i="index">
```
Comment (CrisVega08) [inline src/app/screens/book-list/book-list.component.html]: is unnecessary this ng-template delete

File: src/app/screens/book-list/book-list.component.html [hunk 2]
```diff
@@ -1 +1,9 @@
...
+    <div class="list-item">
```
Comment (CrisVega08) [inline src/app/screens/book-list/book-list.component.html]:
```suggestion
    <div *ngFor="let book of books" class="list-item">
```

File: src/app/services/book.service.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+import { environment } from '@env/environment';
+import { catchError } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
```
Comment (CrisVega08) [inline src/app/services/book.service.ts]: Are you sure of providing this service on all app or only in auth.module?

File: src/app/services/book.service.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+})
+export class BookService {
+  private apiUrl = `${environment.apiUrl}/books`;
+
+  constructor(private http: HttpClient) { }
+
+  getBooks (): Observable<any> {
```
Comment (CrisVega08) [inline src/app/services/book.service.ts]:
```suggestion
  getBooks(): Observable<any> {
```

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/screens/book-list/book-list.component.ts [hunk 5]
```diff
@@ -1,15 +1,21 @@
...
+  books;
 
-  constructor() { }
+  constructor(private bookService: BookService) { }
 
   ngOnInit(): void {
+    this.bookService.getBooks()
+      .subscribe(resp => {
+        this.books = resp.page;
+      })
```
Comment (betoobernal) [inline src/app/screens/book-list/book-list.component.ts]: use destructuring
```suggestion
    this.bookService.getBooks()
      .subscribe(({page}) => {
        this.books = page;
      })
```

File: src/app/token.interceptor.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  constructor(private store: LocalStorageService, private userService: UserService) {}
+
+  intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+    if (this.userService.isLoggedIn()) {
+      request = request.clone({
+        setHeaders: {
+          'access-token': this.store.get('access-token'),
+          client: this.store.get('client'),
+          uid: this.store.get('uid')
+        }
```
Comment (betoobernal) [inline src/app/token.interceptor.ts]: nice

File: src/app/screens/book-list/book-list.component.html [hunk 7]
```diff
@@ -1 +1,7 @@
-<p>book-list works!</p>
+<div class="list-container">
+  <div *ngFor="let book of books"  class="list-item">
```
Comment (CrisVega08) [inline src/app/screens/book-list/book-list.component.html:2]:
```suggestion
  <div *ngFor="let book of books" class="list-item">
```

File: src/app/services/book.service.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+import { HttpClient } from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+import { environment } from '@env/environment';
+import { catchError } from 'rxjs/operators';
+
+@Injectable()
+export class BookService {
+  private apiUrl = `${environment.apiUrl}/books`;
+
+  constructor(private http: HttpClient) { }
```
Comment (CrisVega08) [inline src/app/services/book.service.ts:11]:
```suggestion
  constructor(private http: HttpClient) {}
```

**Review by CrisVega08 — APPROVED** [review]

**Review by betoobernal — APPROVED** [review]

**Pull request merged**